src/OrchardCore/OrchardCore.Search.Elasticsearch.Core/Services/IndexingBackgroundTask.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.BackgroundTasks;
+
+namespace OrchardCore.Search.Elasticsearch.Core.Services
+{
+    /// <summary>
+    /// This background task will index content items using.
+    /// </summary>
+    /// <remarks>
+    /// This services is only registered from OrchardCore.Search.Elasticsearch.Worker feature.
+    /// </remarks>
+    [BackgroundTask(Schedule = "* * * * *", Description = "Update Elasticsearch indexes.")]
+    public class IndexingBackgroundTask : IBackgroundTask
```

Comment:
Is there a case when we don't want this background task to run? (I don't understand the need for a separate feature.)

Reply:
I understand that it is better to have it enabled at all times but someone could want to keep it also disabled. At the same time the background task itself can be disabled so I think we could have it as part of the main feature.

Reply:
I have adjusted the LockTimeout to 1 second and LockExpiration to 1 minute. This way it is close to the minimum of a CronTab schedule allowed. If someone needs to reindex everything then it will do it in multiple background tasks. Generally, it should need to index only a few items at the same time so it should be always shorter than 1 min to execute a batch indexing. I tested with a database that has 40k items to index and it took 4:40 with Lucene so it should be even faster with Elasticsearch.

Reply:
Does your thumbs up to my previous message mean an agreement to my "Do I understand correctly..." question?

Reply:
Concerning nodes, I'm not an expert with Elasticsearch. I believe that you can deploy nodes with different strategies; but here is the documentation about that: 
https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-node.html

And yes, the background task in Elasticsearch should be taking care of sending indexing tasks that have failed for some reason. The Handler will do the work for a user interaction but it could be canceled for different reasons (network failure).
The background task will then trigger itself every period of time to call the ProcessContentItemsAsync method of the ElasticSearchIndexingService which is responsible for indexing batches of Content Items (100 at a time). The reason why we should keep it as a separate feature is because of the fact that it could be implemented differently by someone else. Also, the background task is not a "fire and forget" scheduler which means we are adding it for now but it could be done better. Also, having a background task like this in OC can be an overhead for perf that someone doesn't want to cope with. The background task could be for example executed only once per week or per year if someone would want it to be that way. But it can also be executed every minute which can be an unnecessary overhead.

There is currently no difference between the Lucene and Elasticsearch background tasks. Except the Elasticsearch one has a locking mechanism because it is a distributed service.

Reply:
By multi-node setup I meant having multiple web server nodes, all connecting to the same Elasticsearch instance (like it happens with SQL DBs or AWS S3). Is my understanding above thus correct?

What you describe is not about this, but thank you, noted, and I have no objection to keep the bg task in a separate feature.

Reply:
Yes in a multi-instance deployment of Orchard Core each instance will trigger its own background task and then could also potentially execute duplicated tasks. This is why we added a locking on the background task itself. But, what I was saying is that generally, it will send only a few content items to index with the background task. Only those that have failed to index. So, the locking on the background task will prevent executing too long tasks with it.

Reply:
OK then.